src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,86 @@
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const todo = useAppSelector((state: RootState) => state.currentTodo);
 
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+  if (!todo) {
+    return null;
+  }
+
+  const { title, id, completed, userId } = todo;
+
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  useEffect(() => {
+    const fetchUser = async () => {
+      setIsLoading(true);
+      try {
+        const fetchedUser = await getUser(userId);
 
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button type="button" className="delete" data-cy="modal-close" />
-        </header>
+        setTimeout(() => {
+          setUser(fetchedUser);
+          setIsLoading(false);
+        }, 300);
+      } catch (error) {
+        // eslint-disable-next-line no-console
+        console.error('Ошибка при загрузке данных пользователя:', error);
+        setIsLoading(false);
+      }
+    };
 
-        <div className="modal-card-body">
-          <p className="block" data-cy="modal-title">
-            fugiat veniam minus
-          </p>
+    fetchUser();
+  }, [userId]);
 
-          <p className="block" data-cy="modal-user">
-            {/* For not completed */}
-            <strong className="has-text-danger">Planned</strong>
+  return (
+    <div className="modal is-active" data-cy="modal">
+      <div className="modal-background" />
+      {isLoading ? (
+        <Loader />
+      ) : (
+        <div className="modal-card">
+          <header className="modal-card-head">
+            <div
+              className="modal-card-title has-text-weight-medium"
+              data-cy="modal-header"
+            >
+              {id ? `Todo #${id}` : 'Todo'}
+            </div>
 
-            {/* For completed */}
-            <strong className="has-text-success">Done</strong>
-            {' by '}
-            <a href="mailto:Sincere@april.biz">Leanne Graham</a>
-          </p>
+            <button
+              type="button"
+              className="delete"
+              data-cy="modal-close"
+              onClick={() => onClose(false)}
+            />
+          </header>
+
+          <div className="modal-card-body">
+            <p className="block" data-cy="modal-title">
+              {title ? `${title}` : null}
+            </p>
+
+            <p className="block" data-cy="modal-user">
+              {completed ? (
+                <strong className="has-text-success">Done</strong>
+              ) : (
+                <strong className="has-text-danger">Planned</strong>
+              )}
+              {' by '}
+              {user && <a href={`mailto:${user.email}`}>{user.name}</a>}
+            </p>
```

Comment:
This is a code quality issue. The code inside this block is a bit hard to read due to the conditional rendering and string concatenation. It might be more readable if you extract this into a separate function or component. Also, consider using template literals for string concatenation, it's more readable and less error-prone.